Sources/core/objects/Human.cpp

```diff
@@ -82,6 +84,274 @@ bool Human::TryMove(Dir direct)
         {
             PlaySoundIfVisible("glass_step.ogg", GetOwner().Id());
         }
+        if (pulled_object_)
+        {
+            if (!CanTouch(pulled_object_))
```

Comment:
I checked it and you are right, though the double click shift while pulling still exists. I will make it later.